From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dueris <jedimastertoothless@hotmail.com>
Date: Tue, 19 Aug 2025 16:09:26 -0700
Subject: [PATCH] Disable Useless Particles

Original license: GPL-3.0-only
Original project: https://github.com/CraftCanvasMC/Canvas

It is recommended to enable these options, as the client displays most of these particles already, so the server-side particle logic is not needed

diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index 94b0d4f70dfd9ed5eaf7df7923b7c67c8c617b8a..d9139bb7856f41446a3e9fc260e3f59f61a47507 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -990,7 +990,7 @@ public abstract class Entity
         }
 
         if (this instanceof ServerPlayer) this.handlePortal(); // CraftBukkit - Moved up to postTick
-        if (this.canSpawnSprintParticle()) {
+        if (!org.dreeam.leaf.config.modules.opt.ReduceUselessPackets.disableUselessParticles && this.canSpawnSprintParticle()) { // Leaf - Disable Useless Particles
             this.spawnSprintParticle();
         }
 
@@ -2360,6 +2360,7 @@ public abstract class Entity
 
         float yt = Mth.floor(this.getY());
 
+        if (!org.dreeam.leaf.config.modules.opt.ReduceUselessPackets.disableUselessParticles) { // Leaf - Disable Useless Particles
         for (int i = 0; i < 1.0F + this.dimensions.width() * 20.0F; i++) {
             double xo = (this.random.nextDouble() * 2.0 - 1.0) * this.dimensions.width();
             double zo = (this.random.nextDouble() * 2.0 - 1.0) * this.dimensions.width();
@@ -2374,6 +2375,7 @@ public abstract class Entity
             double zo = (this.random.nextDouble() * 2.0 - 1.0) * this.dimensions.width();
             this.level().addParticle(ParticleTypes.SPLASH, this.getX() + xo, yt + 1.0F, this.getZ() + zo, movement.x, movement.y, movement.z);
         }
+        } // Leaf - Disable Useless Particles
 
         this.gameEvent(GameEvent.SPLASH);
     }
@@ -4096,6 +4098,7 @@ public abstract class Entity
     }
 
     protected static void sendBubbleColumnParticles(final Level level, final BlockPos pos) {
+        if (org.dreeam.leaf.config.modules.opt.ReduceUselessPackets.disableUselessParticles) return; // Leaf - Disable Useless Particles
         if (level instanceof ServerLevel serverLevel) {
             RandomSource random = level.getRandom();
 
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index ade14363a168a3ab2650d6b0d832a431f9639150..af775113bebde3b49ded7d870e75419514b6e361 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -1002,7 +1002,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
             }
 
             List<ParticleOptions> particles = this.entityData.get(DATA_EFFECT_PARTICLES);
-            if (!particles.isEmpty()) {
+            if (!org.dreeam.leaf.config.modules.opt.ReduceUselessPackets.disableUselessParticles && !particles.isEmpty()) { // Leaf - Disable Useless Particles
                 boolean isAmbient = this.entityData.get(DATA_EFFECT_AMBIENCE_ID);
                 int bound = this.isInvisible() ? 15 : 4;
                 int ambientFactor = isAmbient ? 5 : 1;
@@ -2842,7 +2842,7 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
                 this.swapHandItems();
                 break;
             case 60:
-                this.makePoofParticles();
+                if (!org.dreeam.leaf.config.modules.opt.ReduceUselessPackets.disableUselessParticles) this.makePoofParticles(); // Leaf - Disable Useless Particles
                 break;
             case 65:
                 this.breakItem(this.getItemBySlot(EquipmentSlot.BODY));
